homeassistant/components/solax/sensor.py

```diff
@@ -151,6 +156,12 @@ def __init__(
         self._attr_native_unit_of_measurement = unit
         self._attr_state_class = state_class
         self._attr_device_class = device_class
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, serial)},
```

Comment:
Why isn't the unique id the identifier?

Reply:
Because the unique id contains the name of the sensor. By using the serial number all entities will be linked to the right device

Reply:
Ok, was perhaps more wondering why serial wasn't used as unique id but perhaps that's included in there to make it unique.